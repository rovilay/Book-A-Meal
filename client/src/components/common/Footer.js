import React from 'react';

const Footer = () => (
  <footer className="footer">
    <div>
      <p>Book-A-Meal, copyright &copy; 2018</p>
    </div>
  </footer>
);

export default Footer;
