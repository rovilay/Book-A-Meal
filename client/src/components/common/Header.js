/* eslint jsx-a11y/no-static-element-interactions: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint class-methods-use-this: 0 */
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import classname from 'classnames';

import { emptyCart } from '../../actions/cartActions';
import { setDefaultNav } from '../../actions/navLinksActions';
import { logOutUser } from '../../actions/loginActions';

export class Header extends Component {
  constructor(props) {
    super(props);

    this.onLogOut = this.onLogOut.bind(this);
    this.toggleHam = this.toggleHam.bind(this);
  }

  onLogOut() {
    this.props.emptyCart();
    this.props.setDefaultNav();
    this.props.logOutUser();
    this.props.history.push('/');
  }


  toggleHam() {
    const nav = document.getElementById('nav-menu');
    if (nav && nav.className === 'nav-menu') {
      nav.className = 'nav-menu responsive';
    } else if (nav && nav.className === 'nav-menu responsive') {
      nav.className = 'nav-menu';
    }
  }

  render() {
    const {
      navLinks,
      cart
    } = this.props;

    return (
      <header className="header">
        <NavLink to="/" className="logo" exact>Book-A-Meal</NavLink>
        <a
          className="ham"
          onClick={this.toggleHam}
        >
          <FontAwesome
            id="mobile-menu"
            name="bars"
          />
        </a>
        <div className="nav-menu" id="nav-menu">
          {navLinks &&
            navLinks.map((nav, i) => {
              if (nav.title === 'Cart') {
                return (
                  <NavLink
                    to={nav.link}
                    key={i}
                    className="is-active nav-link"
                  >
                    <FontAwesome
                      name="cart-plus"
                    />
                    <span
                      className={
                        classname('', { 'cart-count': cart.length > 0 })
                      }
                    >
                      ({cart.length})
                    </span>
                  </NavLink>
                );
              }

              if (nav.title === 'Log Out') {
                return (
                  <a
                    className= "is-active nav-link"
                    onClick={this.onLogOut}
                    key={i}
                  >
                    {nav.title}
                  </a>
                );
              }

              return (
                <NavLink
                  to={nav.link}
                  key={i}
                  className="is-active nav-link"
                >
                  {nav.title}
                </NavLink>
              );
            })
          }
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  navLinks: PropTypes.array.isRequired,
  history: PropTypes.object.isRequired,
  emptyCart: PropTypes.func.isRequired,
  setDefaultNav: PropTypes.func.isRequired,
  logOutUser: PropTypes.func.isRequired,
  cart: PropTypes.array.isRequired

};

export const mapStateToProps = state => ({
  navLinks: state.navLinks,
  cart: state.cart.meals
});

/* istanbul ignore next */
const mapDispatchToProps = dispatch => bindActionCreators(
  {
    emptyCart,
    setDefaultNav,
    logOutUser
  },
  dispatch
);

export default connect(
  mapStateToProps, mapDispatchToProps
)(withRouter(Header));
