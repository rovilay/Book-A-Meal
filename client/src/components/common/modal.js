import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

import OrderDetailsTable from '../customerpages/modalTables/orderDetails';
import EditOrderTable from '../customerpages/modalTables/editOrder';

Modal.setAppElement('#root');
const ModalComp = (props) => {
  const {
    isOpen,
    isInfo,
    isEdit,
    content,
    contentLabel
  } = props.modal;

  return (
    <Modal
      isOpen={isOpen}
      contentLabel= {contentLabel}
      overlayClassName="overlay"
      className="modal"
    >
      <div className="closeBtn">
        <button
          type="button"
          onClick={props.hideModal}
          className="btn-col btn-1"
        >
          close
        </button>
      </div>

      {
        (isInfo && content)
        &&
        (
          <OrderDetailsTable
            title={contentLabel}
            content={content}
            {...props}
          />
        )
      }

      {
        (isEdit)
        &&
        (
          <EditOrderTable
            title={contentLabel}
            isEdit={isEdit}
            {...props}
          />
        )
      }
    </Modal>
  );
};

ModalComp.propTypes = {
  modal: PropTypes.object.isRequired,
  setModal: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
};

export default ModalComp;
