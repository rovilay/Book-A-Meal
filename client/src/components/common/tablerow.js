import React, { Component } from 'react';
import propTypes from 'prop-types';

class TableRow extends Component {
  constructor(props) {
    super(props);
    this.props = { data: propTypes.object };
  }
  render() {
    return (
      <tr>
        <td>{this.props.data.id}</td>
        <td>{this.props.data.name}</td>
      </tr>
    );
  }
}

export default TableRow;
