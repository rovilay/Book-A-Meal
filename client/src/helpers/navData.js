const adminNavDefault = [
  {
    title: 'Add Meal',
    link: '/dashboard/addmeal'
  },
  {
    title: 'Orders',
    link: '/dashboard/orders'
  },
  {
    title: 'Log Out',
    link: '/'
  }
];

const adminNav = [
  {
    title: 'Set Menu',
    link: '/dashboard'
  },
  {
    title: 'Orders',
    link: '/dashboard/orders'
  },
  {
    title: 'Log Out',
    link: '/'
  }
];

const customerNav = [
  {
    title: 'Menu',
    link: '/dashboard'
  },
  {
    title: 'Cart',
    link: '/cart'
  },
  {
    title: 'Orders',
    link: '/orders'
  },
  {
    title: 'Log Out',
    link: '/'
  }
];

const navData = {
  customerNav,
  adminNav,
  adminNavDefault
};

export default navData;

