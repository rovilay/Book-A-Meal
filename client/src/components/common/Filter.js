/* eslint jsx-a11y/label-has-for:0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { MonthFilter } from '../../helpers/tableHeadData';

class FilterComp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filter: 'all',
      date: '',
      month: ''
    };

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  handleFormSubmit(e) {
    e.preventDefault();
    this.props.setFilter({ ...this.state });
  }

  render() {
    return (
      <div className="filter">
        <form onSubmit={this.handleFormSubmit}>
          <label htmlFor="filter" className="label">Filter By:</label>
          <div className="input-div">
            <select
              name="filter"
              id="filter"
              onChange={(e) => {
                this.setState({ [e.target.name]: e.target.value });
              }}
            >
              <option value="all">All</option>
              <option value="date">Date</option>
              <option value="month">Month</option>
            </select>
          </div>
          {
            (this.state.filter === 'date')
            &&
            <div className="input-div inner-select">
              <input
                type="date"
                name="date"
                id="filter-date"
                required
                onChange={(e) => {
                  this.setState({ month: '', [e.target.name]: e.target.value });
                }}
              />
            </div>
          }

          {
            (this.state.filter === 'month')
            &&
            <div className="input-div inner-select">
              <select
                name="month"
                id="month"
                required
                onChange={(e) => {
                  this.setState({ date: '', [e.target.name]: e.target.value });
                }}
              >
                <option value="">select month</option>
                {
                  MonthFilter.map((month, i) => (
                    <option
                      key={i}
                      value={month}
                    >
                      {month}
                    </option>
                  ))
                }
              </select>
            </div>
          }

          <button
            type="submit"
            value="Search"
            className="btn-2 filter-btn"
          >
            Filter
          </button>
        </form>
      </div>
    );
  }
}

FilterComp.propTypes = {
  setFilter: PropTypes.func.isRequired,
};

export default FilterComp;
