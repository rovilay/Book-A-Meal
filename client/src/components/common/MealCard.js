import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import sweetAlert from 'sweetalert';
import FontAwesome from 'react-fontawesome';

import { getFromLocalStorage } from '../../helpers/localstorage';

class MealCard extends Component {
  constructor(props) {
    super(props);

    this.handleDeleteMeal = this.handleDeleteMeal.bind(this);
  }

  /**
   * handles delete meal
   * @param {*} event DOM event
   */
  handleDeleteMeal(event) {
    event.preventDefault();
    sweetAlert({
      text: 'Are you sure you want to delete this meal?',
      buttons: true,
      dangerMode: true,
    })
      .then((confirmed) => {
        if (confirmed) {
          this.props.deleteMeal(this.props.mealData.id);
        }
      })
      .catch(err => err);
  }

  render() {
    const {
      mealData,
      addToCart,
      editMeal,
      cart
    } = this.props;

    const mealsInCart = (cart) && cart.map(meal => meal.id);
    const user = getFromLocalStorage('user');

    return (
      <div className="mealCard-container">
        <img className="meal-img" src={mealData.image} alt={mealData.title} />
        <div className="overlay1">
          <span className="meal-title">{mealData.title}</span>
        </div>
        <div className="overlay">
          <div className="meal-desc">
            <h4>{mealData.description}</h4>
          </div>
          {
            (user && user.admin)
            &&
            <div className="meal-desc">{moment(mealData.createdAt).format('LL')}</div>
          }
        </div>

        <div className="meal-label">
          {
            // show for customers and visitor only
            (user === null || (user && !user.admin))
            &&
            <button
              className="responsive-btn-2"
              disabled={(mealsInCart) && mealsInCart.includes(mealData.id)}
              onClick={
                () => {
                  addToCart(mealData);
                }
              }
            >
              {
                (mealsInCart && mealsInCart.includes(mealData.id))
                  ?
                  'Meal In Cart'
                  :
                  'Add To Cart'
              }
            </button>
          }
          {
            // show for admin only
            (user && user.admin)
            &&
            (
              <button
                className="btn-3 box-shadow"
                onClick={() => {
                  editMeal(mealData.id);
                }}
              >
                <FontAwesome
                  name="pencil"
                  size="2x"
                  className="pencil"
                />
              </button>
            )
          }

          {
            // show for admin only
            (user && user.admin)
            &&
            (
              <button
                className="btn-3 box-shadow"
                onClick={this.handleDeleteMeal}
              >
                <FontAwesome
                  name="trash"
                  size="2x"
                  className="trash"
                />
              </button>
            )
          }
          <span className="meal-price">&#8358; {mealData.price}</span>
        </div>
      </div>
    );
  }
}


MealCard.defaultProps = {
  addToCart: undefined,
  deleteMeal: undefined,
  editMeal: undefined,
  cart: undefined
};

MealCard.propTypes = {
  mealData: PropTypes.object.isRequired,
  addToCart: PropTypes.func,
  deleteMeal: PropTypes.func,
  editMeal: PropTypes.func,
  cart: PropTypes.array,
};

export default MealCard;
