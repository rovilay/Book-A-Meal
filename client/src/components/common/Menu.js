import React from 'react';
import PropTypes from 'prop-types';

import MealCard from './MealCard';
import '../../assets/css/mealCard.css';

const Menu = (props) => {
  const { menu } = props;
  return (
<<<<<<< HEAD
    <section className="menu">
      {
        (menu.length > 0)
        &&
        <div className="title" id="title">
          {"Today's Menu"}
        </div>
      }
      {
        (menu.length > 0)
          ?
            <div className="menu-container">
              {
                menu.map(meal => (
                  <MealCard
                    key={meal.id}
                    mealData={meal}
                    {...props}
                  />
                ))
              }
            </div>
          :
            <div className="empty">
                Sorry No Meals Today!
            </div>
=======
    <div className="menu">
      {
        (menu.length > 0)
        &&
        <h1 className="menu-title" id="menu-title">
          Menu
        </h1>
      }
      {
        (menu.length > 0)
        ?
          <div className="menu-container" id="">
            {
              menu.map(meal => (
                <MealCard
                  key={meal.id}
                  mealData={meal}
                  {...props}
                />
              ))
            }
          </div>
        :
          <div className="empty">
              Sorry No Meals Today!
          </div>
>>>>>>> d49f2b57ebed161ee85a43f5828ca57edba505e1
      }
    </section>
  );
};

Menu.propTypes = {
  menu: PropTypes.array.isRequired,
};

export default Menu;
