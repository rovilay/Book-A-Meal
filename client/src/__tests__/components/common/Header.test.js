import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps } from '../../../components/common/Header';
import navData from '../../../helpers/navData'
import { meals } from '../../__mockData__/mealMock';


describe('Header component test', () => {
  const setup = () => {

    const props = {
      setDefaultNav: () => jest.fn(),
      logOutUser: () => jest.fn(),
      emptyCart: () => jest.fn(),
      navLinks: navData.customerNav,
      history: [],
      cart: []
    };

    return shallow(<Header {...props} />)
  };

  it('should render Header correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should call `onLogOut` if logout button is clicked', (done) => {
    const wrapper = setup();

    const onLogOutSpy = jest.spyOn(wrapper.instance(), 'onLogOut');
    wrapper.instance().onLogOut()


    expect(onLogOutSpy).toHaveBeenCalled();

    done();
  });


  it('should call `toggleHam` if hamburger menu is clicked on mobile',
  (done) => {
    const wrapper = setup();

    const ontoggleHamSpy = jest.spyOn(wrapper.instance(), 'toggleHam');
    wrapper.instance().toggleHam()


    expect(ontoggleHamSpy).toHaveBeenCalled();

    done();
  });


  it('should map state to props', () => {
    const initialState = {
      navLinks: [],
      cart: {
        meals: []
      }
    };

    const ownProps = {
      navLinks: navData.adminNav,
      cart: {
        meals
      }
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  })
});