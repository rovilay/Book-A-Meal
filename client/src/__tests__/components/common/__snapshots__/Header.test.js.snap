// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test should map state to props 1`] = `
Object {
  "cart": Array [],
  "navLinks": Array [],
}
`;

exports[`Header component test should render Header correctly 1`] = `
<header
  className="header"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="logo"
    exact={true}
    to="/"
  >
    Book-A-Meal
  </NavLink>
  <a
    className="ham"
    onClick={[Function]}
  >
    <FontAwesome
      id="mobile-menu"
      name="bars"
    />
  </a>
  <div
    className="nav-menu"
    id="nav-menu"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="is-active nav-link"
      key="0"
      to="/dashboard"
    >
      Menu
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="is-active nav-link"
      key="1"
      to="/cart"
    >
      <FontAwesome
        name="cart-plus"
      />
      <span
        className=""
      >
        (
        0
        )
      </span>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="is-active nav-link"
      key="2"
      to="/orders"
    >
      Orders
    </NavLink>
    <a
      className="is-active nav-link"
      key="3"
      onClick={[Function]}
    >
      Log Out
    </a>
  </div>
</header>
`;
